log4j-core/src/main/java/org/apache/logging/log4j/core/async/DefaultAsyncWaitStrategyFactory.java

```diff
@@ -74,6 +74,19 @@ static WaitStrategy createDefaultWaitStrategy(final String propertyName) {
         LOGGER.trace(
                 "DefaultAsyncWaitStrategyFactory creating TimeoutBlockingWaitStrategy(timeout={}, unit=MILLIS)",
                 timeoutMillis);
+        try {
+            // Check for the v 4.x version of the strategy, the version in 3.x is not garbage-free.
+            if (DisruptorUtil.DISRUPTOR_MAJOR_VERSION == 4) {
```

Comment:
Nitpick: I would have preferred `if (...) { try {` rather than `try { if (...) {`. The former makes it clear that the upcoming block is about Disruptor 4.